### PR TITLE
[NT-688] Project events & properties clean up

### PR DESCRIPTION
**Pull request by ifbarrera**
# 📲 What

Cleans up our project events, properties and tests to align with new requirements.

# 🤔 Why

So that we're sending only what is needed for analytics purposes.

# 🛠 How

Lots of moving around and renaming things, deleting un-used props, and adding better test coverage.

# ♿️ Accessibility 

N/A

# 🏎 Performance

N/A

# ✅ Acceptance criteria

- [x] `Project Page Viewed` event fires when the project page is viewed
- [x] `Project Swiped` event fires when the project page controller is swiped to the next or previous project

The following properties are part of the `projectProperties` group, and are sent with all `Project` events:

- [x] `project_backers_count`
- [x] `project_category`
- [x] `project_country`
- [x] `project_creator_uid`
- [x] `project_currency`
- [x] `project_deadline` // time interval
- [x] `project_duration` // in days
- [x] `project_goal`
- [x] `project_hours_remaining` // in hours
- [x] `project_launched_at` // time interval
- [x] `project_location`
- [x] `project_name`
- [x] `project_parent_category`
- [x] `project_percent_raised`
- [x] `project_pid`
- [x] `project_pledged`
- [x] `project_has_video`
- [x] `project_current_pledge_amount_usd`
- [x] `project_goal_usd`
- [x] `project_state`
- [x] `project_static_usd_rate`
- [x] `project_user_has_watched`
- [x] `project_user_is_backer`
- [x] `project_user_is_project_creator`
- [x] `project_comments_count`

# ⏰ TODO

- [x] determine whether adding `project_tags` is a blocker for this PR: we don't currently have access to project tags in our `Project` model
- [x] get clarification on whether `project_is_repeat_creator` is actually sending the desired value

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Library/Tracking/Koala.swift [hunk 1]
```diff
@@ -2052,45 +2042,64 @@ public final class Koala {
...
-  props["duration"] = Int(round(project.dates.deadline - project.dates.launchedAt))
+  props["static_usd_rate"] = project.stats.staticUsdRate
+  props["current_pledge_amount_usd"] = project.stats.pledgedUsd
+  props["goal_usd"] = project.stats.goalUsd
+  props["has_video"] = project.video != nil
 
-  props["category"] = project.category.name
-  props["parent_category"] = project.category.parent?.name
+  let now = dateType.init().date
+  props["hours_remaining"] = project.dates.hoursRemaining(from: now, using: calendar)
```
Comment (justinswart) [inline Library/Tracking/Koala.swift]: For projects that have ended in the past this tracks a negative value. Do we want to `min` that as `0`?

File: Library/Tracking/Koala.swift [hunk 2]
```diff
@@ -2052,45 +2042,64 @@ public final class Koala {
...
+  dateType: DateProtocol.Type = AppEnvironment.current.dateType,
+  calendar: Calendar = AppEnvironment.current.calendar,
+  prefix: String = "project_"
 ) -> [String: Any] {
   var props: [String: Any] = [:]
 
   props["backers_count"] = project.stats.backersCount
+  props["category"] = project.category.name
   props["country"] = project.country.countryCode
+  props["comments_count"] = project.stats.commentsCount
```
Comment (justinswart) [inline Library/Tracking/Koala.swift]: This is `nil` if the project has no comments. Think maybe `0` is better here?

File: Library/Tracking/Koala.swift [hunk 3]
```diff
@@ -2052,45 +2042,64 @@ public final class Koala {
...
+  props["location"] = project.location.name
   props["name"] = project.name
-  props["pledged"] = project.stats.pledged
+  props["pid"] = project.id
+  props["parent_category"] = project.category.parent?.name
   props["percent_raised"] = project.stats.fundingProgress
-  props["has_video"] = project.video != nil
+  props["pledged"] = project.stats.pledged
   props["state"] = project.state.rawValue
-  props["update_count"] = project.stats.updatesCount
```
Comment (justinswart) [inline Library/Tracking/Koala.swift]: Seems strange to remove `update_count`?

File: Library/Tracking/Koala.swift (hunk 1 above)
Comment (ifbarrera) [reply Library/Tracking/Koala.swift]: Good idea, nice catch.

File: Library/Tracking/Koala.swift (hunk 3 above)
Comment (ifbarrera) [reply Library/Tracking/Koala.swift]: Ah seems I missed that that one's been renamed to `updates_count`! Good 👁

**Pull request merged**